flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -428,6 +485,12 @@ private void initializeTableEnvironment() {
 			modules.forEach(tableEnv::loadModule);
 		}
 
+		if (catalogManager != null) {
```

Comment:
I think you can split the logic into 2 groups:
1. initializing this with previous catalog manager
2. initializing this without previous catalog manager

and do a if-else here. Step 3-6 can be extracted to a new init catalog method